### ISSUE TITLE
gnt-backup list fails if one node of the cluster is down.

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#77]
Originally reported of Google Code with ID 21.

```
<b>What steps will reproduce the problem?</b>
1. Configure one cluster with at least 2 nodes.
2. Export instances in the two machines using gnt-backup export
3. Remove one of the nodes 
4. Try to run gnt-backup list

<b>What is the expected output? What do you see instead?</b>
We Should see one message that the node is offline or anything related to
the node.  But The program fails and send the following output
"""
 ./gnt-backup list
caller_connect: could not connect to remote host nodo207.labcluster, reason
[Failure instance: Traceback (failure with no frames):
twisted.internet.error.ConnectError: An error occurred while connecting:
113: No route to host.
]
Node: nodo207.labcluster
Exports:
Traceback (most recent call last):
  File "./gnt-backup", line 159, in ?
    sys.exit(GenericMain(commands))
  File "/usr/lib/python2.4/site-packages/ganeti/cli.py", line 251, in
GenericMain
    result = func(options, args)
  File "./gnt-backup", line 46, in PrintExportList
    for instance_name in exports[node]:
TypeError: iteration over non-sequence
"""

<b>What version of the product are you using? On what operating system?</b>
I am using Ganeti 1.2b3, but i had tested it with ganeti 1.2 and the
problem still exist.

This happens because exports[node] returns the boolean value False, when
the node is offline, and the for structure can't make interation over
non-sequence.  

There is two possible solutions, make exports[node] return ['False'] when
the node is offline, or test the result of exports[node] before making the
interation. i had solved it using the second aproach, and print one message
if the node is offline. 

Attached is the patch that solve this issue.

Best Regards.
Leonardo Rodrigues de Mello
```
Originally added on 2007-12-11 17:31:40 +0000 UTC.

Attachments: 
[gnt-backup.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-21/comment-0/gnt-backup.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
The step 3 to reproduce the problem is wrong. 
The step 3 is to STOP OR SHUTDOWN THE NODE. And not to remove the node from the
cluster. sorry for this error in the report.

best regards.
Leonardo Rodrigues de Mello
```
Originally added on 2007-12-11 19:02:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2007-12-12 10:42:04 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is fixed by svn revision 513, which should also be merged in the 1.2 branch
before the next point release.
```
Originally added on 2008-01-22 08:52:24 +0000 UTC.

Changed State: Started
Added to Milestone: Release1.2

**Comment by Ganeti-Issues-Migrator** [comment]
```
The fix has been released in release 1.2.2
```
Originally added on 2008-01-30 10:31:16 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
